wolang/builtInFunctions.go

```diff
@@ -4,27 +4,31 @@ import (
 	"fmt"
 )
 
-func plus(terms []interface{}) (result interface{}, err error) {
-	result = int(0)
+func plus(terms []DataType) (result Addable, err error) {
+	result = Float{0}
 
 	for _, v := range terms {
-		if val, ok := v.(int); !ok {
-			return nil, fmt.Errorf("error: '%v' is not a number I can add!", v)
+		if t, ok := v.(Addable); !ok {
+			return nil, fmt.Errorf("error: '%v' is not a number I can add!", v.GetValue())
 		} else {
-			result = result.(int) + val
+			if result, err = result.Add(t); err != nil {
```

Comment:
Cool cool, no problem. We can totally merge this PR now. If we want we can always change this behaviour later, also the idea is to use this as a base to experiment so i think this is super!!!